sbi/inference/trainers/npe/npe_base.py

```diff
@@ -569,6 +574,50 @@ def _get_potential_function(
         )
         return potential_fn, theta_transform
 
+    def _check_prior_for_rejection_sampling(
+        self,
+        prior: Optional[Distribution],
+        sample_with: Literal["mcmc", "rejection", "vi", "importance", "direct"],
+        posterior_parameters: Optional[
+            Union[
+                DirectPosteriorParameters,
+                MCMCPosteriorParameters,
+                VIPosteriorParameters,
+                RejectionPosteriorParameters,
+                ImportanceSamplingPosteriorParameters,
+            ]
+        ],
+    ) -> None:
+        """
+        Validates that when using rejection sampling, a prior distribution
+        is explicitly provided.
+
+        Args:
+            prior: Prior distribution.
+            sample_with: The sampling method used. Must be one of
+                "mcmc", "rejection", "vi", "importance", or "direct".
+            posterior_parameters: Configuration for building the posterior.
+        """
+
+        if sample_with == "rejection" and prior is None:
+            raise ValueError(
+                "You passed `sample_with='rejection' but you did not specify a "
+                "`prior`. Until sbi v0.22.0, this was interpreted as directly"
+                " sampling from the posterior. As of sbi v0.23.0, you instead have"
+                " to use `sample_with='direct'` to do so."
```

Comment:
this is good, but there is also the possibility that the user just wants to use rejection sampling with the prior as proposal and just forgot to pass the prior. this should be added. e.g., 

"As of sbi v0.23.0, you either have to pass a prior to perform rejection sampling using the prior as proposal, or to use the posterior as proposal, you have to use a DirectPosterior via `sample_with='direct'."

please adapt here and also in the error below. Or, actually, aren't the two cases overlapping? this message should be shown whenever 

`((sample_with="rejection" or isinstance(posterior_parameters, RejectionPosteriorParameters)) and prior is None)`?

so you could just combine them saying that "you indicated sampling via rejection sampling but you ...".